src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts

```diff
@@ -272,27 +282,41 @@ export class NetworkAreaDiagramViewer {
         if (svg != null) {
             svg.style.cursor = 'grabbing';
         }
-        this.initialPosition = DiagramUtils.getPosition(this.selectedElement); // used for the offset
-        this.edgeAngles = new Map<string, number>();
+        if (!(event as MouseEvent).shiftKey) {
+            this.initialPosition = DiagramUtils.getPosition(
+                this.selectedElement
+            ); // used for the offset
+            this.edgeAngles = new Map<string, number>();
+        }
     }
 
     private drag(event: Event) {
         if (this.selectedElement) {
             event.preventDefault();
-            this.ctm = this.svgDraw?.node.getScreenCTM();
-            const mousePosition = this.getMousePosition(event as MouseEvent);
-            this.updateGraph(mousePosition);
-            this.initialPosition = DiagramUtils.getPosition(
-                this.selectedElement
-            );
+            if (!(event as MouseEvent).shiftKey) {
+                this.ctm = this.svgDraw?.node.getScreenCTM();
+                const mousePosition = this.getMousePosition(
+                    event as MouseEvent
+                );
+                this.updateGraph(mousePosition);
+                this.initialPosition = DiagramUtils.getPosition(
+                    this.selectedElement
+                );
+            }
         }
     }
 
     private endDrag(event: Event) {
         if (this.selectedElement) {
-            const mousePosition = this.getMousePosition(event as MouseEvent);
-            this.updateGraph(mousePosition);
-            this.updateMetadataCallCallback(mousePosition);
+            if (!(event as MouseEvent).shiftKey) {
```

Comment:
same comment, you could use a boolean parameter instead

Reply:
done